### PR TITLE
Add PostgresCell

### PR DESCRIPTION
**Pull request by fabianfett**
### Motivation

In #219 we want to restructure PostgresRow and introduce PostgresRandomAccessRow. To make it very explicit, it would be cool if `PostgresRow` was implementing `Collection` and `PostgresRandomAccessRow` was implementing `RandomAccessCollection`. However to do this we need an Element type. This is what `PostgresCell` shall be.

### Changes

- Add `PostgresCell`
- Remove `PSQLData`

### Result

We have an awesome Element type for PostgresRow.

### CODE REVIEW

File: Sources/PostgresNIO/New/PostgresCell.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+struct PostgresCell: Equatable {
+    var bytes: ByteBuffer?
+    var dataType: PostgresDataType
+    var format: PostgresFormat
+
+    var columnName: String
+    var columnIndex: Int
+
+    /// use this only for testing
```
Comment (gwynne) [inline Sources/PostgresNIO/New/PostgresCell.swift:11]: There are no other initializers and the type is not `Codable` - this is the only way to create one of these. I'm wondering if this is potentially an example of splitting up the PRs _too_ much- if there're missing pieces that you couldn't even add via extension (such as non-delegating initializers), the PR is not complete 😕.

Or the comment may just be outdated, but then what would I make a fuss about? 😉

**Review by gwynne — CHANGES_REQUESTED** [review]
Other than the one concern, I don't see anything that needs changing... is there a more specific reason to keep this PR separate other than general desire to cut down on large PRs? (A worthy goal, to be sure!)

File: Sources/PostgresNIO/New/PostgresCell.swift (hunk 1 above)
Comment (fabianfett) [reply Sources/PostgresNIO/New/PostgresCell.swift]: The comment was outdated. Removed. 🤦‍♂️

**Review by gwynne — APPROVED** [review]
Merge away!

**Pull request merged**